### PR TITLE
Gave the headings to all the files functions

### PR DESCRIPTION
**Pull request by ibilalkayy**
Hey @monstermunchkin

The [previous PR](https://github.com/canonical/lxd/pull/12104) was closed accidentally, so I sent this new one.

Here it contains all the files with headings in their functions and there is no merge conflict also.

Thank you!

### CODE REVIEW

**Comment by monstermunchkin** [comment]
@ibilalkayy Please rebase correctly, so that you don't need the last two commits.

**Comment by ibilalkayy** [comment]
The `lxd/rbac/server.go` file function was added in the `lxd/auth/driver_rbac.go` file and 

Similarly in the `lxd/config/generate/lxddoc.go` file, this new code `func writeDocFile(inputYamlPath, outputTxtPath string) error {}` was updated, that's why I removed the old code `func writeDocFile(path string, yaml *doc) error {}` and updated the one new.

**Comment by ibilalkayy** [comment]
@monstermunchkin Now they are fine.

File: lxd/auth/driver_rbac.go [hunk 1]
```diff
@@ -474,7 +474,12 @@ func (r *rbac) syncPermissions(username string) error {
 	return nil
 }
 
+<<<<<<< HEAD:lxd/auth/driver_rbac.go
 func (r *rbac) postResources(updates []rbacResource, removals []string, force bool) error {
+=======
+// Sends resource updates and removals to the API server and handles conflicts by syncing projects.
+func (r *Server) postResources(updates []rbacResource, removals []string, force bool) error {
```
Comment (ibilalkayy) [inline lxd/auth/driver_rbac.go:481]: @monstermunchkin  Due to this function, I added another commit. Even after rebasing, it was still giving me an error. I want you to remove this function and add `func (r *rbac) postResources(updates []rbacResource, removals []string, force bool) error {}` that is matched with the LXD updated code.

**Review by ibilalkayy — COMMENTED** [review]
@monstermunchkin I want you to review these two changes because I didn't update them and this is the reason that so many tests are failing.

Even after rebasing, the two functions were not updating and they were requiring commit and that's why I added two commits but then you said and I removed those commits and stashed those changes.

File: lxd/auth/driver_rbac.go (hunk 1 above)
Comment (monstermunchkin) [reply lxd/auth/driver_rbac.go:481]: You'll need to manually fix this change.

**Comment by monstermunchkin** [comment]
> @monstermunchkin I want you to review these two changes because I didn't update them and this is the reason that so many tests are failing.
> 
> Even after rebasing, the two functions were not updating and they were requiring commit and that's why I added two commits but then you said and I removed those commits and stashed those changes.

You removed the commits, but left the conflicts in the file. Those conflicts cause LXD not to build due to malformed go code. Once the conflicts are resolved, it will build again. You will need to edit those files to fix the issue.

**Comment by ibilalkayy** [comment]
@monstermunchkin I modified the functions in two files that were giving the errors.

**Comment by ibilalkayy** [comment]
Although I created the two commits previously but after rebasing, they were not set up according to the updated LXD code. That's why I manually set them again and matched them with the LXD new code with two new commits.

**Comment by monstermunchkin** [comment]
> I don't know how to modify the last commits again after adding them.

With `git rebase -i`. That will allow you to edit older commits. [Here's a tutorial](https://www.bryanbraun.com/2019/02/23/editing-a-commit-in-an-interactive-rebase/#the-interactive-rebase).

**Comment by ibilalkayy** [comment]
Hey @monstermunchkin now I understood the problem and the problem is that the two files or the two new commits that I added were not present in previous month version. 

You can check the previous month release of the `lxd/lxd/auth` directory but when I upstreamed the code, it added those two new files also.

Although I searched the whole previous commit history but those two files were not present. So it is not a matter of editing the old commits. It's a matter of adding the two new commits and that's why I added them.

**Comment by ibilalkayy** [comment]
Now the commits are edited and the number of commits are the previous one.

**Comment by monstermunchkin** [comment]
@ibilalkayy The static analysis tests are failing. Please sort out those issues.

**Comment by ibilalkayy** [comment]
@monstermunchkin There are some files that need the endpoint and it will take me some time to edit them.

**Comment by ibilalkayy** [comment]
Hey @monstermunchkin, I solved the first check by giving the endpoint to headings but now it is giving me another error that is `Tests / System tests (1.18.x, cluster, zfs) (pull_request)`. I don't know where this problem is present

**Comment by monstermunchkin** [comment]
> Hey @monstermunchkin, I solved the first check by giving the endpoint to headings but now it is giving me another error that is `Tests / System tests (1.18.x, cluster, zfs) (pull_request)`. I don't know where this problem is present

The zfs test failures are unrelated to this PR. You just need to make sure the static check succeeds.

**Comment by ibilalkayy** [comment]
@monstermunchkin All the checks are successfully passed except this ZFS one.

**Comment by monstermunchkin** [comment]
@ibilalkayy Please rebase your changes onto main, instead of adding a merge commit.

**Comment by ibilalkayy** [comment]
@monstermunchkin Now I have rebased it and the merge conflict is also removed.

**Comment by tomponline** [comment]
Hi @ibilalkayy please can you rebase again so that we can ensure the docs test passes, thanks

**Comment by ibilalkayy** [comment]
@tomponline, @monstermunchkin  Finally all the checks are passed and there is no merge conflict.

File: lxd/endpoints/object.go [hunk 2]
```diff
@@ -12,6 +12,8 @@ import (
 	"github.com/canonical/lxd/shared/logger"
 )
 
+// StorageBucketsCreateListener establishes a secure TCP listener for storage bucket communication,
```
Comment (monstermunchkin) [inline lxd/endpoints/object.go:15]:
```suggestion
// storageBucketsCreateListener establishes a secure TCP listener for storage bucket communication,
```

File: lxd/endpoints/pprof.go [hunk 3]
```diff
@@ -12,6 +12,7 @@ import (
 	"github.com/canonical/lxd/shared/logger"
 )
 
+// PprofCreateServer creates a standalone HTTP server dedicated to serving Go's pprof debugging requests.
```
Comment (monstermunchkin) [inline lxd/endpoints/pprof.go:15]:
```suggestion
// pprofCreateServer creates a standalone HTTP server dedicated to serving Go's pprof debugging requests.
```

**Comment by ibilalkayy** [comment]
@monstermunchkin It will take me some time to look at each heading.

**Comment by ibilalkayy** [comment]
@monstermunchkin The comments are now corrected and they are changed from upper case to lower case as they are written in the functions.

**Comment by monstermunchkin** [comment]
@ibilalkayy ping me once you want me to review this PR again.

**Comment by ibilalkayy** [comment]
@monstermunchkin I have changed the function names in the comments from upper case to lower case and all the checks are passed successfully. Now you can review them.

**Comment by tomponline** [comment]
This needs a rebase please

**Comment by ibilalkayy** [comment]
@monstermunchkin and @tomponline What do I need to change?

**Comment by monstermunchkin** [comment]
> @monstermunchkin and @tomponline What do I need to change?

`lxd/daemon.go` has conflicts which need to be solved.

**Comment by tomponline** [comment]
This is proving very difficult to review in time to avoid conflicts.
Are you able to reduce the size of the changes to just functions that you feel could benefit from a doc block? 
I'd like to get this PR merged. Thanks

**Comment by ibilalkayy** [comment]
@tomponline I didn't understand what you said. Tell me briefly.

**Comment by tomponline** [comment]
Please can you rebase this at least so the PR passes tests

**Comment by tomponline** [comment]
Hi @ibilalkayy we've found this PR too large to review.

Also we've recently changed to require the Canonical CLA to be signed see https://github.com/canonical/lxd/blob/main/CONTRIBUTING.md#license-and-copyright

Please can open separate PRs, with commits per change restricted to each package you're changing, this should balance out the number of PRs vs the number of commits per PR and allow us to review the changes in a reasonable time.

Thanks

**Pull request closed without merging**